linkerd/trace-context/src/lib.rs

```diff
@@ -16,10 +16,10 @@ use thiserror::Error;
 
 const SPAN_ID_LEN: usize = 8;
 
-#[derive(Debug, Default)]
+#[derive(Debug, Default, Clone)]
```

Comment:
This doesn't need to be Clone. We can instead use a reference to `TraceContext`. hex decoder accepts anything that impl `AsRef<u8>`, we can use that for the `Id` type.